### DISCUSSION TITLE
Advice for the best quality

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#2965] [Q&A]
Hello, everyone, I 've tested the Fooocus for faceswap with config below:
model: SDXL: juggernautXL_v8Rundiffusion.safetensors
step: 15

The text prompt: (full body: 1.4), Edgy Rocker, Beautiful girl, picturesque cityscape, magnificent modern architecture, beautiful streets, sunlight streaming, historic architecture, modern magnificent skyscraper, incredible inspiring city

negative_prompt: (nsfw, naked, nude, deformed iris, deformed pupils, semi-realistic, cgi, 3d, render, sketch, cartoon, drawing, anime, mutated hands and fingers:1.4), (deformed, distorted, disfigured:1.3), poorly drawn, bad anatomy, wrong anatomy, extra limb, missing limb, floating limbs, disconnected limbs, mutation, mutated, ugly, disgusting, amputation, Naked, Nude, NSFW, nake, nude, disfigured, kitsch, ugly, oversaturated, grain, low-res, Deformed, blurry, bad anatomy, disfigured, poorly drawn face, mutation, mutated, extra limb, ugly, poorly drawn hands, missing limb, blurry, floating limbs, disconnected limbs, malformed hands, blur, out of focus, long neck, long body, ugly, disgusting, poorly drawn, childish, mutilated, mangled, text, blurry, b&w, monochrome, conjoined twins, multiple heads, extra legs, extra arms, meme, deformed, elongated, twisted fingers, strabismus, closed eyes, blurred, watermark  '

Result: 
![20cd63d5-294b-412c-adca-ea66e590fe52](https://github.com/lllyasviel/Fooocus/assets/91205658/86bac7e3-8f70-44f6-983d-d2ce028d5caa)

The image generation took 15s

Problems: There is a 12% chance of bad legs, arms, limbs, eyes, teeth. The body feels distorted.
I want to depress this into 5%

Another example: 
The text prompt: (full body: 1.4), Edgy Rocker, Beautiful girl, European vacation, Picturesque street, historic stone buildings, colorful flowers, cute cafes, awnings, people, blue sky, sunlight, windows in the buildings, flowerpots.
Result: 
![20cd63d5-294b-412c-adca-ea66e590fe52](https://github.com/lllyasviel/Fooocus/assets/91205658/9a2d445d-cfe8-42f9-94db-2a7e598f29e3)

![2d743778-9674-4d9d-a8dc-a1346b5f43bb](https://github.com/lllyasviel/Fooocus/assets/91205658/b7a22289-04d3-4098-812e-6e68b842ac4a)

Text: (full body: 1.4),  Edgy Rocker, Beautiful girl, smiling, magnificent luxury fashion, luxury handbag, New York City, Winter, illuminated lights on Fifth Avenue, elegant luxury storefronts, festive christmas winter decorations, warm glow, snow-dusted sidewalks. Christmas tree shines, glittering lights, skyscrapers, enchanting, festive ambiance.

Result: 

![12127ab0-e19d-4b11-925c-812521ca7e99](https://github.com/lllyasviel/Fooocus/assets/91205658/8c58ebfd-2212-4a31-a462-f88145f083a2)

As you can see this results, some of the background features are missing and eyes, faces are uneasy.
I saw some of the articles to handle uneasy eyes, faces but I don't know to add codebase to automatically handle this errors.
I would like to hear any advice from the expert on Fooocus.

Thank you very much.

**Comment by abzaloff** [comment]
You pay too little attention to the character, and too much attention to the environment. Describe the character in more detail, focusing on the face and other details. Fewer negative clues. In Fooocus, you can do without them or use a minimum of them. And 15 steps won't give you a great result.
